libs/product/src/selectors/product-entities/product-entities.selectors.ts

```diff
@@ -85,6 +86,15 @@ const createProductEntitiesSelectors = <T extends DaffProduct>(): DaffProductEnt
 		}
 	);
 
+	const selectProductDiscountPercent = createSelector(
+		selectProductEntities,
+		(products, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+
+			return (product.discount && product.discount.percent) || 0;
```

Comment:
can you add an optional chaining TODO here?

Reply:
done